beacon_node/lighthouse_network/src/rpc/mod.rs

```diff
@@ -408,16 +405,29 @@ where
                                     format!("Wait {:?}", wait_time).into(),
                                 ),
                             );
+                            return;
                         }
+                        // No rate limiting, continue.
+                        Ok(_) => {}
                     }
-                } else {
-                    // No rate limiting, send the event to the user
-                    self.events.push(ToSwarm::GenerateEvent(RPCMessage {
+                }
+                // If we received a Ping, we queue a Pong response.
+                if let InboundRequest::Ping(_) = req {
+                    trace!(self.log, "Received Ping, queueing Pong";"connection_id" => %conn_id, "peer_id" => %peer_id);
+                    self.send_response(
                         peer_id,
-                        conn_id,
-                        message: Ok(RPCReceived::Request(id, req)),
-                    }))
+                        (conn_id, id),
+                        RPCCodedResponse::Success(RPCResponse::Pong(Ping {
+                            data: self.seq_number,
+                        })),
+                    );
                 }
+
+                self.events.push(ToSwarm::GenerateEvent(RPCMessage {
```

Comment:
If we are handling the Ping response here, do we still need to send the PING request to the swarm?

Reply:
yeah, I'd say in general yes, to allow the other behaviours or even the swarm to react to `Ping`s

Reply:
Yeah we need to let the peer manager know about the ping so that it can send metadata requests if the seq number is updated.